Add hybrid lambda pkg

downloadico opened the pull request.

downloadico committed:
hybrid-lambda: add hybrid-lambda package

downloadico committed:
I thought that I was committing an updated version.  oops!

alalazo commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,41 @@
+    def autoreconf(self, spec, prefix):
+        autoreconf('--install', '--verbose', '--force')
+
+    def build(self, spec, prefix):
+        with working_dir('src'):
+            make
```
Can this be avoided by setting the appropriate `configure_directory` or `build_directory` attributes?

downloadico replied:
Yes!  Thanks, I'll remove this shortly!

adamjstewart replied:
In fact, as is, this line won't do anything. You can't call a function without parentheses. But yes, setting one of those variables should work.

alalazo commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,41 @@
+    depends_on('boost', type='build')
+    depends_on('cppunit', type='build')
+
+    def autoreconf(self, spec, prefix):
+        autoreconf('--install', '--verbose', '--force')
```
Is the default method not working correctly?

downloadico replied:
yes, it is!  Oops!  I'll remove this shortly!

alalazo commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,41 @@
+        with working_dir('src'):
+            make
+
+    def install(self, spec, prefix):
+        filter_file(r'INSTALL = /bin/install -c',
+                    'INSTALL = /bin/install -C', 'Makefile')
+        make('install')
```
Would this:
```suggestion
    @run_after('configure')
    def change_install_option_in_makefile(self):
        filter_file(r'INSTALL = /bin/install -c',
                    'INSTALL = /bin/install -C', 'Makefile')
```
work?

downloadico replied:
Almost...  I'll commit a fix that uses this mechanism shortly!

alalazo commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,41 @@
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('m4', type='build')
+    depends_on('boost', type='build')
```
Boost is very likely a `build` and `link` dependency, unless you use boost as a build tool
```suggestion
    depends_on('boost')
```

alalazo commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,41 @@
+    depends_on('libtool', type='build')
+    depends_on('m4', type='build')
+    depends_on('boost', type='build')
+    depends_on('cppunit', type='build')
```
@adamjstewart Should we do something like:
```suggestion
    depends_on('cppunit', type='test')
```
for unit testing frameworks?

alalazo commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,41 @@
+
+
+class HybridLambda(AutotoolsPackage):
+    """Hybrid-Lambda is a software package that can simulate gene trees
+       within a rooted species network or a rooted species tree under the
+       coalescent process. The main feature of this program is that users can
+       choose to use the standard Kingman coalescent process, which produces
```
If you leave an empty line and start with a short paragraph you'll get both a short and a long description of the software in different contexts:
```suggestion
    """Hybrid-Lambda is a software package that can simulate gene trees
       within a rooted species network or a rooted species tree under the
       coalescent process. 
       
       The main feature of this program is that users can
       choose to use the standard Kingman coalescent process, which produces
```

downloadico replied:
Thanks for the suggestion!

alalazo reviewed (changes requested):
A few questions

downloadico committed:
Apply suggestions from code review

Adding suggestions from alalanzo

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

downloadico committed:
set the build directory to 'src'
removed autoreconf method
replaced install method with change_install_option_in_makefile to be run
 after configure method

adamjstewart commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,39 @@
+    homepage = "https://github.com/hybridLambda/hybrid-Lambda"
+    git      = "https://github.com/hybridLambda/hybrid-Lambda.git"
+
+    version('dev', submodules=True)
```
```suggestion
    version('develop', submodules=True)
```

downloadico replied:
looks good, dev was from their build docs, but develop works as well

adamjstewart replied:
Just to be clear, `version('master', submodules=True)` will download the same version as the previous two. If you don't specify a `branch`, it will download the default branch. The first arg is just the version name that Spack calls it.

adamjstewart commented on var/spack/repos/builtin/packages/hybrid-lambda/package.py:
```diff
@@ -0,0 +1,39 @@
+    def change_install_option_in_makefile(self):
+        with working_dir('src'):
+            filter_file(r'INSTALL = /bin/install -c',
+                        'INSTALL = /bin/install -C', 'Makefile')
```
Is this actually necessary? From the macOS man page, it seems like `-c` and `-C` are almost identical.

downloadico committed:
fixed style errors (from a suggestion; I didn't think to flake8 on it. oops)

downloadico committed:
Update var/spack/repos/builtin/packages/hybrid-lambda/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

downloadico reviewed (commented):
I believe it's working and much prettier now

downloadico commented:
Yes.  The behavior is different on Linux.

Jim

On Mon, 11 Jan 2021, Adam J. Stewart wrote:

> @adamjstewart commented on this pull request.
>
>
>
>> +    version('dev', submodules=True)
> +
> +    depends_on('autoconf', type='build')
> +    depends_on('automake', type='build')
> +    depends_on('libtool', type='build')
> +    depends_on('m4', type='build')
> +    depends_on('boost')
> +    depends_on('cppunit', type='test')
> +
> +    build_directory = 'src'
> +
> +    @run_after('configure')
> +    def change_install_option_in_makefile(self):
> +        with working_dir('src'):
> +            filter_file(r'INSTALL = /bin/install -c',
> +                        'INSTALL = /bin/install -C', 'Makefile')
>
> Is this actually necessary? From the macOS man page, it seems like `-c` and `-C` are almost identical.
>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/spack/spack/pull/20754#pullrequestreview-565645718

James E. Prewett                    Jim@Prewett.org download@hpc.unm.edu
Systems Team Leader           LoGS: http://www.hpc.unm.edu/~download/LoGS/
Designated Security Officer         OpenPGP key: pub 1024D/31816D93
HPC Systems Engineer III   UNM HPC  505.277.8210

downloadico commented:
Thanks for the clarification Adam!

Jim

On Mon, 11 Jan 2021, Adam J. Stewart wrote:

> @adamjstewart commented on this pull request.
>
>
>
>> +class HybridLambda(AutotoolsPackage):
> +    """Hybrid-Lambda is a software package that can simulate gene trees
> +       within a rooted species network or a rooted species tree under the
> +       coalescent process.
> +
> +       The main feature of this program is that users can
> +       choose to use the standard Kingman coalescent process, which produces
> +       bifurcating genealogies, or two other Lambda coalescent processes,
> +       which produce multifurcating genealogies. The other feature is that
> +       hybrid sim uses extended Newick formatted strings to make it easier to
> +       represent hybridization events between species."""
> +
> +    homepage = "https://github.com/hybridLambda/hybrid-Lambda"
> +    git      = "https://github.com/hybridLambda/hybrid-Lambda.git"
> +
> +    version('dev', submodules=True)
>
> Just to be clear, `version('master', submodules=True)` will download the same version as the previous two. If you don't specify a `branch`, it will download the default branch. The first arg is just the version name that Spack calls it.
>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/spack/spack/pull/20754#discussion_r555383962

James E. Prewett                    Jim@Prewett.org download@hpc.unm.edu
Systems Team Leader           LoGS: http://www.hpc.unm.edu/~download/LoGS/
Designated Security Officer         OpenPGP key: pub 1024D/31816D93
HPC Systems Engineer III   UNM HPC  505.277.8210
